File: app/src/main/java/ru/divizdev/coinrate/presentation/coinRateList/presenter/CoinRateListView.java
package ru.divizdev.coinrate.presentation.coinRateList.presenter;


import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.AddToEndSingleStrategy;
import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;
import ru.divizdev.coinrate.presentation.entities.CoinRateUI;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface CoinRateListView extends MvpView {
    void showLoadingProgress(Boolean isView);

    void showCoinRateList(List<CoinRateUI> list);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showErrorLoading();

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showDialogAbout();

    @StateStrategyType(OneExecutionStateStrategy.class)
    void navToDetail(CoinRateUI coinRateUI);
}

File: app/src/main/java/ru/divizdev/coinrate/presentation/coinRateList/presenter/CoinRateListView.kt
package ru.divizdev.coinrate.presentation.coinRateList.presenter

import moxy.viewstate.strategy.StateStrategyType
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.MvpView
import ru.divizdev.coinrate.presentation.entities.CoinRateUI
import moxy.viewstate.strategy.OneExecutionStateStrategy

@StateStrategyType(AddToEndSingleStrategy::class)
interface CoinRateListView : MvpView {
    fun showLoadingProgress(isView: Boolean?)
    fun showCoinRateList(list: List<CoinRateUI?>?)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showErrorLoading()

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showDialogAbout()

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun navToDetail(coinRateUI: CoinRateUI?)
}